src/main/java/store/config/InitialDataLoader.java

```diff
@@ -0,0 +1,106 @@
+package store.config;
+
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayDeque;
+import java.util.List;
+import java.util.Queue;
+import store.error.ErrorMessage;
+import store.error.exception.InvalidFailInitialDataLoadException;
+import store.product.domain.Product;
+import store.product.domain.ProductInfo;
+import store.product.domain.ProductRepository;
+import store.promotion.domain.Promotion;
+import store.promotion.domain.PromotionDateTime;
+import store.promotion.domain.PromotionRepository;
+import store.promotion.domain.PromotionType;
+import store.util.StringParser;
+
+public class InitialDataLoader {
+
+    private static final String PATH_PROMOTIONS = "promotions.md";
+    private static final String PATH_PRODUCTS = "products.md";
+    private static final String DELIMITER_COMMA = ",";
+    private final PromotionRepository promotionRepository;
+    private final ProductRepository productRepository;
+    private Long promotionSequence = 1L;
+    private Long productSequence = 1L;
```

Comment:
p4

```suggestion
    private final AtomicLong promotionSequence = new AtomicLong(1);
    private final AtomicLong productSequence = new AtomicLong(1);
```

`AtomicLong`을 사용하면 멀티스레드에도 대응이 되고 `final Long id = productSequence++` 처럼  후위 증감 연산자 대신 `final Long id = productSequence.getAndIncrement()` 메서드를 통해서 관리가 가능해집니다.

Reply:
AtomicLong이 그런 기능이 있었군요 ㅎㅎ

한번도 사용해본적이 없는데 이번 리팩토링때 적용해볼게요!